lib/ht_members.rb

```diff
@@ -1,17 +1,151 @@
 # frozen_string_literal: true
 
-# WIP: cache of information about HathiTrust members
+require "mysql2"
+require "dotenv"
+
+Dotenv.load(".env")
+
+=begin
+
+Cache of information about HathiTrust members
+The data may be mocked if you pass it in to new(),
+otherwise it will try to read it from db, and if that fails
+it will use canned data (CANNED_DATA).
+
+@mocked and @canned are set accordingly and readable, so you can
+tell where the data is coming from
+
+Usage:
+
+  htm = HTMembers.new()
+  puts "using canned data? #{htm.canned}"
+  cc = htm.get("yale")["country_code"]
+  wt = htm.get("harvard")["weight"]
+
+Or if you want to provide your own mock data:
+
+  htm = HTMembers.new({
+    "haverford" => {
+      "country_code" => "us",
+      "weight"       => 0.67,
+    }
+  })
+  puts "using canned data? #{htm.canned}"
+  puts "using mocked data? #{htm.mocked}"
+  htm.get("haverford")["country_code"]
+  htm.get("haverford")["weight"]
+
+=end
+
 class HTMembers
+  def self.template(cc, wt, sym)
+    {
+      "country_code" => cc,
+      "weight"       => wt,
+      "oclc_sym"     => sym,
+    }
+  end
 
-  def initialize(*members)
-    @members = members.empty? ? load_from_db : members
+  # If we get more cases like universityofcalifornia then this might
+  # look better as a series of SELECT ... UNION SELECT ....
+  SQL = %w<
+    SELECT DISTINCT 
+    IF(
+      mapto_inst_id='universityofcalifornia',
+      inst_id, 
+      mapto_inst_id
+    ) AS member_id,
+    country_code,
+    weight,
+    oclc_sym
+    FROM ht_institutions
+    WHERE enabled = '1'
+  >.join(' ')
+
+  CANNED_DATA = {
```

Comment:
What's the use case for having the canned data rather than just passing it in, if one doesn't want to read it from the database?

Reply:
I can see wanting to have sample data for development purposes, but that seems like perhaps it should be the responsibility of whatever constructs `HTMembers` to provide?